server/server.js

```diff
@@ -150,31 +150,138 @@ app.get("/auth/callback", (req, res) => {
       grant_type: "authorization_code",
     },
     headers: {
-      Authorization:
-        "Basic " +
-        Buffer.from(CLIENT_ID + ":" + CLIENT_SECRET).toString("base64"),
+      Authorization: "Basic " + Buffer.from(CLIENT_ID + ":" + CLIENT_SECRET).toString("base64"),
       "Content-Type": "application/x-www-form-urlencoded",
     },
     json: true,
   };
 
-  // Request the access and refresh tokens
   request.post(authOptions, async (error, response, body) => {
     if (!error && response.statusCode === 200) {
-      const { access_token, refresh_token } = body; // Extract tokens from response
+      const { access_token, refresh_token } = body;
 
-      // Optionally save tokens in the database if needed
+      // Log userId to ensure it's correctly extracted
+      console.log("Extracted userId:", userId);
 
-      // Redirect to the frontend with tokens
-      res.redirect(
-        `http://localhost:5173/home?access_token=${access_token}&refresh_token=${refresh_token}`
-      );
+      await getAndStoreTopTracks(access_token);
+
+      res.redirect(`http://localhost:5173/home?access_token=${access_token}&refresh_token=${refresh_token}`);
     } else {
-      res.status(400).json({ error: "Invalid token" }); // Handle errors
+      res.status(400).json({ error: "Invalid token" });
     }
   });
 });
 
+
+ //Endpoint to get the recommendation factors
+
+app.post("/save-recommendation", async (req, res) => {
+  const { userId, location, weather, placeTypes, expression } = req.body;
+  console.log(req.body);
+  try {
+    const parsedUserId = parseInt(userId, 10);
+    const recommendation = await prisma.recommendation.create({
+      data: {
+        userId: parsedUserId,
+        location: location || "",
+        weather: weather || "", 
+        placeTypes: placeTypes || "",
+        expression: expression || "",   
+      },
+    });
+    res.status(201).json(recommendation);
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ error: "Failed to save recommendation" });
+  }
+});
```

Comment:
Is this part the only new snippet after PR #4?  Please land PR #4 then we can land this.

I see you have many duplicate changes in #4 and #5. You can create a PR on the top of an existing PR if you are working on the same files.  In that case, only the change after #4 will be highlighted.

Creating a PR from the main only applies when you are working on two different things when they are independent to each other.